columnflow/tasks/calibration.py

```diff
@@ -104,6 +104,10 @@ def run(self):
         reader_targets = self.calibrator_inst.run_setup(reqs["calibrator"], inputs["calibrator"])
         n_ext = len(reader_targets)
 
+        # determine the maximum chunk size
+        chunk_sizes = (dep.max_chunk_size for dep in self.calibrator_inst.walk_deps(include_self=True))
+        chunk_size = min((s for s in chunk_sizes if isinstance(s, int)), default=None)
```

Comment:
These two lines are repeated for every task. I feel like this could be a (cached) property of the TaskArrayFunction Class, which could be used directly in Line 140.

Reply:
Then we also have to make `max_chunk_size` an attribute of `TaskArrayFunction` instead of the inheriting CSPs. But I agree, this is a rather generic feature that should live in the parent class.

Reply:
Added in last commit. I guess this requires a re-review.